### PR TITLE
fix: merge tab issues

### LINKED ISSUE #12508: [Bug]: [deploy preview] the conflict error is displayed even when there is no conflict
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The open repo button and merge conflict text is displayed in the merge tab even when when there is no conflict
image

### Steps To Reproduce

1. add a new application
2. connect the application to git
3. now do a initial commit and push
4. now click on the merge tab
5. The open repo and conflict error is displayed before the branch selection

### Public Sample App

_No response_

### Version

deploy preview

### PR DESCRIPTION
**Pull request by IAmAnubhavSaini**
## Description

Refactor `ConflictInfo` component to make it testable.

Fixes #12508 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Tests attached.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix-merge-tab 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.39 **(0.04)** | 37.78 **(0.04)** | 36.01 **(0.07)** | 56.61 **(0.04)**
 :green_circle: | app/client/src/ce/constants/messages.ts | 77.33 **(0.1)** | 100 **(0)** | 32.96 **(0.28)** | 82.04 **(0)**
 :green_circle: | app/client/src/components/ads/Button.tsx | 90.97 **(0)** | 76.52 **(0.87)** | 95.56 **(0)** | 91.55 **(0)**
 :green_circle: | app/client/src/components/ads/Text.tsx | 93.44 **(3.28)** | 84.38 **(12.5)** | 100 **(0)** | 93.33 **(3.33)**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/ConflictInfo.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/InfoWrapper.tsx** | **100** | **75** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/Link.tsx** | **95.24** | **55.56** | **75** | **94.74**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 83.61 **(-1.64)** | 61.76 **(-2.95)** | 73.33 **(0)** | 88.24 **(-2.35)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/components/MergeStatus.tsx [hunk 1]
```diff
@@ -42,10 +40,9 @@ function MergeStatus({
       return (
         <Flex>
           <Wrapper>
-            <CheckLine color={Colors.GREEN} size={18} />
             <Text
               color={Colors.GREEN}
-              style={{ marginLeft: 8, alignSelf: "center" }}
+              style={{ alignSelf: "center" }}
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/gitSync/components/MergeStatus.tsx:45]: any specific reason to use inline style here?

File: app/client/src/pages/Editor/gitSync/components/MergeStatus.tsx [hunk 2]
```diff
@@ -59,10 +56,9 @@ function MergeStatus({
       return (
         <Flex>
           <Wrapper>
-            <ErrorWarning color={Colors.CRIMSON} size={18} />
             <Text
               color={Colors.CRIMSON}
-              style={{ marginLeft: 8, alignSelf: "center" }}
+              style={{ alignSelf: "center" }}
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/gitSync/components/MergeStatus.tsx:61]: same as above

**Review by ankitakinger — APPROVED** [review]
LGTM

**Comment by IAmAnubhavSaini** [comment]
Failing test:
Screenshot 2022-04-13 at 1 04 04 PM

**Comment by IAmAnubhavSaini** [comment]
Failing test:
Screenshot 2022-04-13 at 1 04 04 PM

**Pull request merged**